interface/ceed-operator.c

```diff
@@ -233,6 +233,43 @@ int CeedOperatorGetActiveBasis(CeedOperator op, CeedBasis *active_basis) {
   return CEED_ERROR_SUCCESS;
 }
 
+/**
+  @brief Find the active vector bases for a non-composite CeedOperator
+
+  @param[in] op                CeedOperator to find active bases for
+  @param[out] active_basis_in  Bases for active input vectors
+  @param[out] active_basis_out Bases for active output vectors
+
+  @return An error code: 0 - success, otherwise - failure
+
+  @ ref Developer
+**/
+int CeedOperatorGetActiveBases(CeedOperator op, CeedBasis **active_basis_in, CeedBasis **active_basis_out) {
+  Ceed ceed;
+  CeedCall(CeedOperatorGetCeed(op, &ceed));
+
+  if (op->is_composite) {
+    // LCOV_EXCL_START
+    return CeedError(ceed, CEED_ERROR_MINOR, "composite operator is not supported");
+    // LCOV_EXCL_STOP
+  }
+  CeedCall(CeedCalloc(op->qf->num_input_fields, &op->active_basis_in));
+  CeedCall(CeedCalloc(op->qf->num_output_fields, &op->active_basis_out));
+  for (CeedInt i = 0; i < op->qf->num_input_fields; i++) {
+    if (op->input_fields[i]->vec == CEED_VECTOR_ACTIVE) {
```

Comment:
I don't think we want these NULL spaces in this - it will make these arrays harder to use

Reply:
Right now, for an operator with the inputs

`[interp basis_1, passive qdata, grad basis_1, interp basis_2]`

this array will hold

`[basis_1, NULL, basis_1, basis_2]`

That's sort of awkward compared to how this array will be used. For assembly, we loop over active eval modes like this

`[interp basis_1, grad basis_1, interp basis_2]`

Reply:
I see, thanks